insteonplm/devices/__init__.py

```diff
@@ -671,22 +672,26 @@ def receive_message(self, msg):
                 _LOGGER.debug('DA queue: %s',
                               self._sent_msg_wait_for_directACK)
                 _LOGGER.debug('Message ACK with no callback')
-        if (hasattr(msg, 'flags') and
-                hasattr(msg.flags, 'isDirectACK') and
-                msg.flags.isDirectACK):
-            _LOGGER.debug('Got Direct ACK message')
-            if self._send_msg_lock.locked():
-                self._directACK_received_queue.put_nowait(msg)
-            else:
-                _LOGGER.debug('But Direct ACK not expected')
+
         if not self._is_duplicate(msg):
+            if (hasattr(msg, 'flags') and
```

Comment:
This appears to be the core change. Can you help me understand why checking if a direct ack is a duplicate first helps?

Reply:
I was trying to leave the original use of _is_duplicate() intact since I didn't fully understand the motivation.  It is possible to merge this logic with the logic in the "else" but I took the safer way out.